Bluewire.Tools.Runner/FindTickets/ResolveTicketsFromSemanticVersions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Bluewire.Common.GitWrapper;
+using Bluewire.Common.GitWrapper.Model;
+using Bluewire.Conventions;
+using Bluewire.Tools.Runner.Shared;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Bluewire.Tools.Runner.FindTickets
+{
+    internal class ResolveTicketsFromSemanticVersions : ITicketsResolutionJob
+    {
+        private string endSemanticVersion;
+        private string startSemanticVersion;
+
+        public ResolveTicketsFromSemanticVersions(string startSemanticVersion, string endSemanticVersion)
+        {
+            this.startSemanticVersion = startSemanticVersion;
+            this.endSemanticVersion = endSemanticVersion;
+        }
+
+        public async Task<string[]> ResolveTickets(GitSession session, Common.GitWrapper.GitRepository repository)
+        {
+            var startCommitJob = new FindCommits.ResolveCommitFromSemanticVersion(startSemanticVersion);
+            var startBuilds = await startCommitJob.ResolveCommits(session, repository);
+            var startBuild = Build.SelectHighestPriorityResult(startBuilds);
+
+            Ref startCommit = startBuild.Commit;
+            Ref endCommit = await FindEndCommit(session, repository, startBuild);
+
+            var includeCommits = await session.ReadLog(repository, new LogOptions(), new Difference(startCommit, endCommit));
+            var excludeCommits = await session.ReadLog(repository, new LogOptions(), new Difference(endCommit, startCommit));
+
+            var includeTicketsStrings = includeCommits.SelectMany(c => Patterns.TicketIdentifier.Matches(c.Message).OfType<Match>().Select(m => m.Value));
+            var excludeTicketsStrings = excludeCommits.SelectMany(c => Patterns.TicketIdentifier.Matches(c.Message).OfType<Match>().Select(m => m.Value));
+
+            return includeTicketsStrings.Except(excludeTicketsStrings).ToArray();
+        }
+
+        private async Task<Ref> FindEndCommit(GitSession session, Common.GitWrapper.GitRepository repository, Build startBuild)

```

Comment:
I'd actually be inclined to have the caller worry about working this out, especially since the debug message is ambiguous about which branch HEAD (and where the HEAD actually is at the time). `ResolveTickets` could then take a pair of unambiguous `Build` structures.

The various methods used to work this stuff out could reasonably live in the same class, though.